R/zchunk_LA100.IEA_downscale_ctry.R

```diff
@@ -0,0 +1,308 @@
+      # the USSR and separated countries thereafter, resulting in unrealistic fuel shares.
+      USSR_YUG_SPLIT_YEAR <- 1990
+      USSR_YUG_YEARS <- HISTORICAL_YEARS[HISTORICAL_YEARS < USSR_YUG_SPLIT_YEAR]
+      POST_USSR_YUG_YEARS_IEA <- HISTORICAL_YEARS[HISTORICAL_YEARS >= USSR_YUG_SPLIT_YEAR]
+      L100.USSR_Yug <- filter(L100.IEAcomposite, COUNTRY %in% c("Former Soviet Union (if no detail)",
+                                                                "Former Yugoslavia (if no detail)"))
+
+      # Re-map the "if no detail" forms of coal in the historical years prior to the relevant
+      # coal types for matching with the more recent years (89-107)
+      NO_DETAIL_COAL_YEARS <- as.character(intersect(1971:1977, HISTORICAL_YEARS))
+      if(length(NO_DETAIL_COAL_YEARS)) {
+        # Hard coal needs to be split proportionally between coking coal and other bituminous coal
+        # in order to minimize bias from different country-wise shares of the two fuel types.
+        # Note that anthracite is not considered in these regions.
+        # NOTE: using round() to avoid NA's for any values whose base value is >1e6. This seems to work
+        prod <- L100.USSR_Yug$PRODUCT
+        prod_obc <- prod == "Other bituminous coal"
+        prod_hcind <- prod == "Hard coal (if no detail)"
+        prod_cc <- prod == "Coking coal"
+        L100.USSR_Yug[prod_obc, NO_DETAIL_COAL_YEARS] <-
+          L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] *
+          round(L100.USSR_Yug$`1978`[prod_obc] /
+                  (L100.USSR_Yug$`1978`[prod_cc] + L100.USSR_Yug$`1978`[prod_obc] + 1e-3), digits = 2)
+        L100.USSR_Yug[prod_cc, NO_DETAIL_COAL_YEARS ] <-
+          L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] - L100.USSR_Yug[prod_obc, NO_DETAIL_COAL_YEARS ]
+        L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] <- 0
+
+        # Brown coal is simpler, as lignite is the only relevant fuel (104-119)
+        # (sub-bituminous coal is not considered in these regions)
+        prod_bcind <- prod == "Brown coal (if no detail)"
+        L100.USSR_Yug[prod == "Lignite", NO_DETAIL_COAL_YEARS ] <- L100.USSR_Yug[prod_bcind, NO_DETAIL_COAL_YEARS ]
+        L100.USSR_Yug[prod_bcind, NO_DETAIL_COAL_YEARS ] <- 0
+      }
+
+      # Isolate 1990 data for FSU and Yugoslavia; this will be used to back-project individual country shares
+      fsu_yug_composite_entries <- IEA_ctry_composite$IEA_ctry %in% c("Former Soviet Union (if no detail)", "Former Yugoslavia (if no detail)")
+      L100.IEAsingle %>%
+        filter(iso %in% IEA_ctry_composite$iso[fsu_yug_composite_entries]) %>%
+        left_join_keep_first_only(select(IEA_ctry_composite, iso, IEA_ctry), by = "iso") %>%
+        rename(IEAcomposite = IEA_ctry) ->
+        L100.USSR_Yug_ctry
+
+      L100.USSR_Yug_ctry %>%
+        filter(paste(PRODUCT, FLOW) %in% paste(IEA_product_downscaling$PRODUCT, IEA_product_downscaling$FLOW)) ->
```

Comment:
Wouldn't be better to do this as a `semi_join`?  (It would certainly be clearer; I spent a good minute or two figuring out what this was supposed to do.)